### PR TITLE
3309

### PR DESCRIPTION
**Pull request by carlangas159**
Export de vcard non-fonctionnel 3309

Allow download vcard to a user with relationship.

### CODE REVIEW

File: main/social/vcard_export.php [hunk 1]
```diff
@@ -24,6 +23,14 @@
 // Return User Info to vCard Export
 $userInfo = api_get_user_info($userId, true, false, true);
 
+/* Get the relationship between current user and vcard user */
+$currentUserId = api_get_user_id();
+$hasRelation = (int) (SocialManager::get_relation_between_contacts($currentUserId, $userId));
+if ($hasRelation == 0) {
```
Comment (jmontoyaa) [inline main/social/vcard_export.php:29]: Better to use:

```
if (0 === $hasRelation) {
}
```

File: main/social/vcard_export.php [hunk 2]
```diff
@@ -24,6 +23,14 @@
 // Return User Info to vCard Export
 $userInfo = api_get_user_info($userId, true, false, true);
 
+/* Get the relationship between current user and vcard user */
+$currentUserId = api_get_user_id();
+$hasRelation = (int) (SocialManager::get_relation_between_contacts($currentUserId, $userId));
```
Comment (jmontoyaa) [inline main/social/vcard_export.php:28]: Don't add "(int)" better make sure that SocialManager::get_relation_between_contacts returns an int value doing the cast inside the function.

File: main/social/vcard_export.php (hunk 1 above)
Comment (carlangas159) [reply main/social/vcard_export.php:29]: indeed, the function get_relation_between_contacts returns the id of the record or the USER_UNKNOWN constant, for security I set the int in the case that row ['id'] returns a string

File: main/social/vcard_export.php (hunk 2 above)
Comment (carlangas159) [reply main/social/vcard_export.php:28]: indeed, the function get_relation_between_contacts returns the id of the record or the USER_UNKNOWN constant, for security I set the int in the case that row ['id'] returns a string

Comment (jmontoyaa) [reply main/social/vcard_export.php:28]: Yes, exactly.

**Pull request closed without merging**